src/app/main/component/events/services/events.service.ts

```diff
@@ -11,9 +11,36 @@ export class EventsService implements OnDestroy {
   public currentForm: PagePreviewDTO;
   private backEnd = environment.backendLink;
   private destroyed$: ReplaySubject<any> = new ReplaySubject<any>(1);
+  private isAddressFillSubject: BehaviorSubject<boolean[]> = new BehaviorSubject<boolean[]>([]);
 
   constructor(private http: HttpClient) {}
 
+  public setIsAddressFill(dates: DateEvent[], submit?: boolean, init?: boolean, check?: boolean, ind?: number): void {
+    const currentValues = this.isAddressFillSubject.getValue();
+    let newArray;
+
+    if (init) {
+      newArray = [];
+    } else if (submit) {
+      newArray = dates.map((nextValue) => !(nextValue.coordinatesDto.latitude || nextValue.onlineLink));
+    } else if (check && ind !== undefined) {
+      currentValues[ind] = currentValues[ind] === undefined ? false : !(dates[ind].coordinatesDto.latitude || dates[ind].onlineLink);
```

Comment:
!currentValues[ind]

Reply:
I tried to execute the function with !currentValues[ind] but unfortunately, that breaks the logic of the function. Because when the value === undefined, the new value must be false without being able to become true